### PR TITLE
SSH Slave timeout and retries

### PR DESCRIPTION
**Pull request by jsmickey**
Logic for SSH slaves to reconnect automatically

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: templates/default/sv-jenkins-run.erb [hunk 1]
```diff
@@ -9,7 +9,7 @@
 exec 2>&1
 cd <%= node['jenkins']['master']['home'] %>
 touch jenkins.start
-exec chpst -u <%= node['jenkins']['master']['user'] %> -U <%= node['jenkins']['master']['user'] %> \
+exec chpst -u <%= node['jenkins']['master']['user'] %>:jenkins:docker:rvm -U <%= node['jenkins']['master']['user'] %>:jenkins:docker:rvm \
```
Comment (schisamo) [inline templates/default/sv-jenkins-run.erb]: This change should not be part of this PR. I believe the issue will be fixed as part of https://github.com/chef-cookbooks/jenkins/pull/390.

**Comment by schisamo** [comment]
@jsmickey Be sure to rebase this branch onto master. I just merged some changes that should make the Travis CI execution of the integration tests more stable. Also be sure to add some integration test coverage for your changes!

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 2 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by sterago** [comment]
Any progress on this? Is there a chance that this cookbook supports configuring timeout and retries for slaves anytime soon?

**Comment by jsmickey** [comment]
Pull Request is here: https://github.com/chef-cookbooks/jenkins/pull/425

Not sure on the status of this cookbook.  We have it working in production from this fork: https://github.com/jsmickey/jenkins
